src/__tests__/posthog-core.js

```diff
@@ -127,6 +128,44 @@ describe('posthog core', () => {
             expect(console.error).toHaveBeenCalledWith('[PostHog.js]', 'No event name provided to posthog.capture')
         })
 
+        it('respects opt_out_useragent_filter (default: false)', () => {
+            const originalUseragent = globals.userAgent
+            // eslint-disable-next-line no-import-assign
```

Comment:
Would have preferred to use [jest.spyOn](https://jestjs.io/docs/jest-object#jestspyonobject-methodname) but could not get that to work within `posthog-core.js`. Instead we override the global & make sure to reset it